HelpSource/scdoc.js

```diff
@@ -193,3 +193,35 @@ function fixTOC() {
         set_up_toc();
     }
 }
+
+// Set up a QWebChannel for communicating with C++ IDE objects. The main app publishes a handle
+// to the help browser at "helpBrowser" and made globally available here after the page and
+// WebSocket have loaded.
+function setUpWebChannel() {
+    if (location.search != "")
+        var baseUrl = (/[?&]webChannelBaseUrl=([A-Za-z0-9\-:/\.]+)/.exec(location.search)[1]);
+    else
+        var baseUrl = "ws://localhost:12345";
+
+    var socket = new WebSocket(baseUrl);
+    socket.onclose = function() { };
+    socket.onerror = function(error) {
+        console.error("WebChannel error: " + error);
+    };
+    socket.onopen = function() {
+        new QWebChannel(socket, function(channel) {
+            // make help browser globally accessible
+            window.helpBrowser = channel.objects.helpBrowser;
+        });
+    }
+}
+
+// Fixes the TOC and sets up the QWebChannel for communicating back to C++ code
+function didLoadRenderedWindow() {
+    fixTOC();
+
+    // Check that webchannel.js was loaded
+    if(typeof QWebChannel != "undefined") {
```

Comment:
`!==` is preferred in js instead of `!=` by convention. because less bugs and weird behaviour.